hilti/runtime/include/util.h

```diff
@@ -16,14 +16,37 @@
 #include <utility>
 #include <vector>
 
-#include <hilti/rt/autogen/config.h>
 #include <hilti/rt/exception.h>
 #include <hilti/rt/filesystem.h>
 #include <hilti/rt/result.h>
 #include <hilti/rt/types/set_fwd.h>
 #include <hilti/rt/types/time.h>
 #include <hilti/rt/types/vector_fwd.h>
 
+#define HILTI_RT_ENUM_WITH_DEFAULT(name, default_, ...)                                                                \
+    struct name {                                                                                                      \
+        enum Value : int64_t { __VA_ARGS__ };                                                                          \
+        constexpr name(int64_t value = default_) noexcept : _value(value) {}                                           \
+        friend name Enum(Value value) { return name(value); }                                                          \
+        friend constexpr bool operator==(const name& a, const name& b) noexcept { return a.value() == b.value(); }     \
+        friend constexpr bool operator!=(const name& a, const name& b) noexcept { return ! (a == b); }                 \
+        friend constexpr bool operator<(const name& a, const name& b) noexcept { return a.value() < b.value(); }       \
+        constexpr int64_t value() const { return _value; }                                                             \
+        int64_t _value;                                                                                                \
+    }
+
+#define HILTI_RT_ENUM(name, ...)                                                                                       \
+    struct name {                                                                                                      \
+        enum Value : int64_t { __VA_ARGS__ };                                                                          \
+        constexpr name(int64_t value = Undef) noexcept : _value(value) {}                                              \
```

Comment:
This requires the user of the macro to have provide  `Undef` label, right? Would be worth a comment, I think. Or alternatively, we could just go with one version of the macro and always require a default (which often then would be `Undef`). Fine either way for me.

Reply:
I'll add Doxygen comment blocks to both macros. I'll keep both of them so we have a way to quickly find uses of deviating enum types, but I'll change `HILTI_RT_ENUM` to instead just call `HILTI_RT_ENUM_WITH_DEFAULT` instead to reduce duplication.